test/deriver/test_default.ml

```diff
@@ -56,6 +56,21 @@ type ('a, 'b) parametrized_variant =
   | C of 'b option
 [@@deriving default]
 
+type poly_variant_without_arg = [`A | `B]
```

Comment:
Is it worth making this one `B of int` to make sure the variants with args do not accidentally override?

Reply:
We can definitely change the test case but can you be a little bit more specific about the behaviour we wouldn't want here?

Reply:
Not a big change - just make the variant

    [`A | `B of int]

Just in case the "having argument"-ness of `B` overrides `A` as the one that is used for the default.

Reply:
Ok I'll change that!